packages/block-editor/src/hooks/color.js

```diff
@@ -276,6 +277,47 @@ export function ColorEdit( props ) {
 	);
 }
 
+/**
+ * This adds inline styles for color palette colors.
+ * Ideally, this is not needed and themes should load their palettes on the editor.
+ *
+ * @param  {Function} BlockListBlock Original component
+ * @return {Function}                Wrapped component
+ */
+export const withColorPaletteStyles = createHigherOrderComponent(
+	( BlockListBlock ) => ( props ) => {
+		const { name, attributes } = props;
+		const { backgroundColor, textColor } = attributes;
+		const { colors } = useSelect( ( select ) => {
+			return select( 'core/block-editor' ).getSettings();
+		}, [] );
+
+		if ( ! hasColorSupport( name ) || ! backgroundColor || ! textColor ) {
```

Comment:
This condition may be wrong. If text color is set but a background color is not set we may not be applying the text color style. I'm able to reproduce this issue while testing with the paragraph block and 2019 theme.

Reply:
Good catch